src/ddmd/traits.d

```diff
@@ -370,6 +373,90 @@ extern (C++) Expression pointerBitmap(TraitsExp e)
     return ale;
 }
 
+/**
+* implements __traits(externDmangle, qualified.name, FunctionType)
+*
+* combines a fully qualified symbol name with a function type and
+*  returns an alias to the resulting new symbol. It is neither merged with
+*  existing symbols nor added to symbol lookup tables.
+*
+* Params:
+*  e  = traits expression with two arguments:
+*       1. a string expression with a sequence of identifiers seperated by '.'
+*       2. a function type
+*  sc = current evaluation scope
+*
+* Returns: symbol expression with the new symbol
+*/
+extern (C++) Expression externDmangle(TraitsExp e, Scope* sc)
+{
+    if (!e.args || e.args.dim != 2)
+    {
+        error(e.loc, "a (name,type) pair expected for trait pointerBitmap");
+        return new ErrorExp();
+    }
+
+    auto ex = isExpression((*e.args)[0]);
+    if (!ex)
+    {
+        e.error("expression expected as first argument of __traits %s", e.ident.toChars());
+        return new ErrorExp();
+    }
+    ex = ex.ctfeInterpret();
+
+    StringExp se = ex.toStringExp();
+    if (!se || se.len == 0)
+    {
+        e.error("string expected as first argument of __traits %s instead of %s", e.ident.toChars(), ex.toChars());
+        return new ErrorExp();
+    }
+    se = se.toUTF8(sc);
+
+    Type t = getType((*e.args)[1]);
+    if (t && t.ty == Tpointer && t.nextOf().ty == Tfunction)
+        t = t.nextOf();
+    if (!t || t.ty != Tfunction)
+    {
+        error(e.loc, "%s is not a function type", (*e.args)[1].toChars());
```

Comment:
A better error message might give some context, such as "the second argument to __traits(externDemangle) must be a function type, not %s"